### ISSUE TITLE
Recognize other "global" patterns, not only "window.x"

### ISSUE THREAD

**Issue by givankin** [eugeneware/deglobalify#2]
How much would it take to support "bad modules" that pollute global namespaces by doing

```
var myfunc = function() { return 42 }
```

or

```
function myfunc() { return 42 }
```

or even 

```
this.myfunc = function() { return 42 }
```

?

All of these are out there in the wild, but none of these can be turned to "good module" using deamdify now. Unfortunately language parsing is Greek to me. Is it feasible?

**Comment by CMCDragonkai** [comment]
I agree. I was trying to deglobalify highlight.js which uses `var hljs =`. But it did not work.

**Comment by dannyfritz** [comment]
This seems like a pretty cool idea. Looks like this transform is using AST via Falafel to do the transformation already. I'm not very familiar with Falafel, but with plain AST, it is possible to detect and then fix.

Probably something like

``` javascript
FunctionExpression: function (node) {
  var parent = node.parent();
  if (parent.type === 'Program') {
    //deglobalify here
  }
}
```
